code/game/objects/items/devices/scanners.dm

```diff
@@ -259,21 +259,32 @@ GENE SCANNER
 
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
+		var/list/damaged = H.get_damaged_bodyparts(1,1)
 		for(var/O in H.internal_organs)
 			var/obj/item/organ/organ = O
 			if((organ.damage > organ.low_threshold)&&(!istype(organ, /obj/item/organ/brain)))
 				to_chat(user, "\t<span class='alert'>[organ.damage > organ.high_threshold ? "Severe" : "Minor"] damaged detected within [organ].</span>")
 		if(advanced && H.has_dna())
 			to_chat(user, "\t<span class='info'>Genetic Stability: [H.dna.stability]%.</span>")
-
-	// Body part damage report
-	if(iscarbon(M) && mode == 1)
-		var/mob/living/carbon/C = M
-		var/list/damaged = C.get_damaged_bodyparts(1,1)
 		if(length(damaged)>0 || oxy_loss>0 || tox_loss>0 || fire_loss>0)
-			to_chat(user, "<span class='info'>\tDamage: <span class='info'><font color='red'>Brute</font></span>-<font color='#FF8000'>Burn</font>-<font color='green'>Toxin</font>-<font color='blue'>Suffocation</font>\n\t\tSpecifics: <font color='red'>[brute_loss]</font>-<font color='#FF8000'>[fire_loss]</font>-<font color='green'>[tox_loss]</font>-<font color='blue'>[oxy_loss]</font></span>")
-			for(var/obj/item/bodypart/org in damaged)
-				to_chat(user, "\t\t<span class='info'>[capitalize(org.name)]: [(org.brute_dam > 0) ? "<font color='red'>[org.brute_dam]</font></span>" : "<font color='red'>0</font>"]-[(org.burn_dam > 0) ? "<font color='#FF8000'>[org.burn_dam]</font>" : "<font color='#FF8000'>0</font>"]")
+			// Body part damage report
+			var/dmgreportlist = "<pre><span class='info'>&#9;Damage:&#9;&#9;\
+				<font color='red'><b>Brute</font></span>&#9;\
+				<font color='orange'>Burn</font>&#9;\
+				<font color='green'>Toxin</font>&#9;\
+				<font color='purple'>Suffocation</b></font><br>\
+
+				&#9;&nbsp;&nbsp;&nbsp;&nbsp;<span class='info'>Overall:&#9;\
+				<font color='red'>[brute_loss]</font>&#9;\
+				<font color='orange'>[fire_loss]</font>&#9;\
+				<font color='green'>[tox_loss]</font>&#9;\
+				<font color='purple'>[oxy_loss]</font></span><br>"
+			for(var/obj/item/bodypart/org in damaged) //head, left arm, right arm, etc.

```

Comment:
Make this a typeless for loop.

Reply:
you sure? this is how it is in both the original coder's PR and what is was before. It works as is, if that's the problem

Reply:
Typeless loops are faster, and this can be made typeless.

Reply:
I'll see what I can do, then

Reply:
I think I just made that exact change to my PR, if that helps.

Reply:
Yep, made my job a lot easier by doing that